### PR TITLE
Distinguish unary from binary

### LINKED ISSUE #462: Overhual binop
Current:
- `def infix` handles all operators: unary and binary

Wanted:
- `def infix` handles all binary operators
- `def fixl`, `def fixr` for all unary operators
- Strict argument count assumption and section syntactic sugar

### PR DESCRIPTION
**Pull request by imkiva**
Fix #462 and removed `bind` keyword

before: `def x => {??} bind looser +` => now: `def x => {??} looser +`
before: `def x => {??} bind { looser a tighter b }` => now: `def x => {??} looser a tighter b`

### CODE REVIEW

**Comment by imkiva** [comment]
We need path type conversion to accept this code

image

File: base/src/main/java/org/aya/tyck/unify/DefEq.java [hunk 1]
```diff
@@ -303,19 +304,47 @@ private boolean doCompareTyped(@NotNull Term type, @NotNull Term lhs, @NotNull T
...
+      default -> false;
+    };
+  }
+
+  private boolean compareCube(@NotNull Cube<Term> lhs, @NotNull Cube<Term> rhs, Sub lr, Sub rl) {
+    lhs.params().zipView(rhs.params()).forEach(x -> {
+      lr.map.put(x._1, new RefTerm(x._2));
+      rl.map.put(x._2, new RefTerm(x._1));
+    });
+    // TODO: let CofThy.propExt uses lr and rl?
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/unify/DefEq.java:334]: Can we just capture these variables in the lambda in propExt?

File: tools/src/main/java/org/aya/util/binop/Assoc.java [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
-  /**
-   * Parenthesis operator fixity.
-   */
-  Infix(true),
-  InfixL(true),
-  InfixR(true),
-  Invalid(false);
+  /** infix */
+  Infix,
+  /** infix, but associates leftly */
```
Comment (ice1000) [inline tools/src/main/java/org/aya/util/binop/Assoc.java:11]: left-associative

File: base/src/main/java/org/aya/tyck/unify/DefEq.java (hunk 1 above)
Comment (imkiva) [reply base/src/main/java/org/aya/tyck/unify/DefEq.java:334]: What do you mean by "capture" (sorry I didn't get it)

Comment (imkiva) [reply base/src/main/java/org/aya/tyck/unify/DefEq.java:334]: We can address this later :)

**Comment by imkiva** [comment]
bors merge

**Pull request merged**